FreeRTOS/Demo/Common/Minimal/TaskNotifyArray.c

```diff
@@ -0,0 +1,1243 @@
+			}
+		}
+
+		xTimeOnEntering = xTaskGetTickCount();
+		xReturned = xTaskNotifyWaitIndexed( uxIndexToTest, notifyUINT32_MAX, 0, &ulNotifiedValue, xTicksToWait );
+		( void ) xReturned; /* Remove compiler warnings in case configASSERT() is not defined. */
+
+		/* Should have blocked for the entire block time. */
+		configASSERT( ( xTaskGetTickCount() - xTimeOnEntering ) >= xTicksToWait );
+		configASSERT( xReturned == pdFAIL );
+		configASSERT( ulNotifiedValue == 0UL );
+		( void ) xReturned; /* Remove compiler warnings in case configASSERT() is not defined. */
+		( void ) ulNotifiedValue;
+
+		/* Clear all the other notifications within the array of task
+		notifications again ready for the next round. */
+		for( uxOtherIndexes = 0; uxOtherIndexes < configTASK_NOTIFICATION_ARRAY_ENTRIES; uxOtherIndexes++ )
+		{
+			if( uxOtherIndexes != uxIndexToTest )
+			{
+				xReturned = xTaskNotifyStateClearIndexed( xTaskToNotify, uxOtherIndexes );
+
+				/* The notification state was set above so expect it to still be
+				set. */
+				configASSERT( xReturned == pdTRUE );
+				( void ) xReturned; /* Remove compiler warnings in case configASSERT() is not defined. */
+			}
+		}
+	}
+
+
+
+	/* ------------------------------------------------------------------------
+	Check no blocking when notifications are pending. */
+	for( uxIndexToTest = 0; uxIndexToTest < configTASK_NOTIFICATION_ARRAY_ENTRIES; uxIndexToTest++ )
+	{
+		/* First notify the task notification at index uxIndexToTest within this
+		task's own array of task notifications - this would not be a normal
+		thing to do and is done here for test purposes only. */
+		xReturned = xTaskNotifyAndQueryIndexed( xTaskToNotify, uxIndexToTest, ulFirstNotifiedConst, eSetValueWithoutOverwrite, &ulPreviousValue );
+
+		/* Even through the 'without overwrite' action was used the update should
+		have been successful. */
+		configASSERT( xReturned == pdPASS );
+		( void ) xReturned; /* Remove compiler warnings in case configASSERT() is not defined. */
+
+		/* No bits should have been pending previously. */
+		configASSERT( ulPreviousValue == 0 );
+		( void ) ulPreviousValue;
+
+		/* The task should now have a notification pending in the task
+		notification at index uxIndexToTest within the task notification array,
+		and so not time out. */
+		xTimeOnEntering = xTaskGetTickCount();
+		xReturned = xTaskNotifyWaitIndexed( uxIndexToTest, notifyUINT32_MAX, 0, &ulNotifiedValue, xTicksToWait );
+		configASSERT( ( xTaskGetTickCount() - xTimeOnEntering ) < xTicksToWait );
+
+		/* The task should have been notified, and the notified value should
+		be equal to ulFirstNotifiedConst. */
+		configASSERT( xReturned == pdPASS );
+		configASSERT( ulNotifiedValue == ulFirstNotifiedConst );
+		( void ) xReturned; /* Remove compiler warnings in case configASSERT() is not defined. */
+		( void ) ulNotifiedValue;
+
+		/* Incremented to show the task is still running. */
+		ulFineCycleCount++;
```

Comment:
I'm guessing we might have repurposed the test loops a little bit while adding more tests... The usage of this variable is quite inconsistent in my opinion, in a sense that some functions called in [```prvNotifiedTask()```](https://github.com/FreeRTOS/FreeRTOS/blob/2d8718e25ac9135699b49d602ccb19e1bf8e9955/FreeRTOS/Demo/Common/Minimal/TaskNotifyArray.c#L174) could increment ```ulFineCycleCount``` for more than once. 

Along with this line, I'll try to remove 
https://github.com/FreeRTOS/FreeRTOS/blob/2d8718e25ac9135699b49d602ccb19e1bf8e9955/FreeRTOS/Demo/Common/Minimal/TaskNotifyArray.c#L845
https://github.com/FreeRTOS/FreeRTOS/blob/2d8718e25ac9135699b49d602ccb19e1bf8e9955/FreeRTOS/Demo/Common/Minimal/TaskNotifyArray.c#L880

If removing these three does not impose issue on this check https://github.com/FreeRTOS/FreeRTOS/blob/2d8718e25ac9135699b49d602ccb19e1bf8e9955/FreeRTOS/Demo/Common/Minimal/TaskNotifyArray.c#L1198 (due to the frequency of this check is called and also number of tasks running), I'll include the change in the next revision. Otherwise, will leave it for now, since this variable does not seem to be functionally critical.